File: 1299-replace-elements-with-greatest-element-on-right-side/1299.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        right_max = -1
        r = len(arr) - 1

        while r >= 0:
            curr = arr[r]
            arr[r] = right_max
            if curr > right_max:
                right_max = curr
            r -= 1

        return arr

File: 1299-replace-elements-with-greatest-element-on-right-side/1299.rs
impl Solution {
    pub fn replace_elements(arr: Vec<i32>) -> Vec<i32> {
        let mut curr = -1;
        let mut right_max = -1;
        let mut res = arr;
    
        for i in (0..res.len()).rev() {
            curr = res[i];
            res[i] = right_max;
             if curr > right_max {
                right_max = curr;
            }
        }
       res
    }
}
